_source/_posts/2022-05-04-angular-microfrontend-deploy.md

```diff
@@ -0,0 +1,286 @@
+
+We'll use the `environments` configuration built into Angular to support both local and deployed environments and define a configuration for serving locally versus production build.
+
+We'll configure serving locally first. Open `projects/shell/src/environments/environment.ts` and add a new property for the micro-frontend remotes.
+
+```ts
+export const environment = {
+  production: false,
+  mfe: {
+    "mfeBasket": "http://localhost:4201",
+    "mfeProfile": "http://localhost:4202"
+  }
+};
+```
+
+Next, we'll use the values in the route definition. Open `projects/shell/src/app/app-routing.module.ts` and update the `remoteEntry` properties to use the new properties in the `environment.ts` file below.
+
+```ts
+import { environment } from '../environments/environment';
+
+const routes: Routes = [
+  { path: '', component: ProductsComponent },
+  {
+    path: 'basket',
+    loadChildren: () => loadRemoteModule({
+      type: 'module',
+      remoteEntry: `${environment.mfe.mfeBasket}/remoteEntry.js`,
+      exposedModule: './Module'
+    }).then(m => m.BasketModule)
+  },
+  {
+    path: 'profile',
+    loadChildren: () => loadRemoteModule({
+      type: 'module',
+      remoteEntry: `${environment.mfe.mfeProfile}/remoteEntry.js`,
+      exposedModule: './Module'
+    }).then(m => m.ProfileModule)
+  },
+  { path: 'login/callback', component: OktaCallbackComponent }
+];
+```
+
+Feel free to serve the project using `npm run run:all` to double-check everything still works.
+
+Now we'll add the configuration for the deployed applications. Open `projects/shell/src/environments/environment.prod.ts` and add the same property for the micro-frontend remotes. Replace the URI with the deployed location.
+
+```ts
+export const environment = {
+  production: true,
+  mfe: {
+    "mfeBasket": "https://{yourVercelDeployPath}.vercel.app",
+    "mfeProfile": "https://{yourVercelDeployPath}.vercel.app"
+  }
+};
+```
+
+Angular automatically replaces the `environment.ts` file to match when you build the project. So if you're serving locally, you'll use the contents from `environment.ts`, and if you run `ng build` to create the release, you'll use the contents from `environment.prod.ts`. 
+
+**Note**: _Vercel has a release promotion mechanism where you first release to a staging environment to conduct release verification, then promote the verified application to prod. With this flow, we'd need to add more complexity to the way we access the URIs. This tutorial will simplify things to only use one environment, but if you need to handle a multi-step release environment configuration, check out the links to blog posts at the end for handling per-environment configuration._
```

Comment:
I typically use NOTE in uppercase for notes. It probably doesn't matter as long as you're consistent in the post. I'm not sure you need italics here. You could also change it to a blockquote if you want it to stand out more. Or convert to AsciiDoc where the admonitions are better! https://docs.asciidoctor.org/asciidoc/latest/blocks/admonitions/

I'd be happy to help convert.